zebra-scan/src/scan.rs

```diff
@@ -0,0 +1,58 @@
+//! The scan task.
+
+use std::time::Duration;
+
+use color_eyre::{eyre::eyre, Report};
+use tower::{buffer::Buffer, util::BoxService, Service, ServiceExt};
+use tracing::info;
+
+use crate::storage::Storage;
+
+type State = Buffer<
+    BoxService<zebra_state::Request, zebra_state::Response, zebra_state::BoxError>,
+    zebra_state::Request,
+>;
+
+/// Wait a few seconds at startup so tip height is always `Some`.
+const INITIAL_WAIT: Duration = Duration::from_secs(10);
+
+/// The amount of time between checking and starting new scans.
+const CHECK_INTERVAL: Duration = Duration::from_secs(10);
+
+/// Start the scan task given state and storage.
+///
+/// - This function is dummy at the moment. It just makes sure we can read the storage and the state.
+/// - Modificatiuons here might have an impact in the `scan_task_starts` test.
+/// - Real scanning code functionality will be added in the future here.
+pub async fn start(mut state: State, storage: Storage) -> Result<(), Report> {
+    // We want to make sure the state has a tip height available before we start scanning.
+    std::thread::sleep(INITIAL_WAIT);
```

Comment:
Using `thread::sleep()` in an async function will pause the whole async executor. It can also lead to compiler errors, because the future is not able to hold some kinds of data over a sleep. I'm not sure if that's the error you're having?

Try using `tokio::time::sleep(...).await` instead:
https://docs.rs/tokio/latest/tokio/time/fn.sleep.html

This also applies to the other sleep.

Reply:
Thank you for that, i don't do async code since a while. The initial sleep is just to make sure we have a tip but it can be removed. I changed it to tokio sleep by now.